src/attribute.c

```diff
@@ -23,3 +23,100 @@ void SetPlayerName(u16 *nameBuffer)
     }
 }
 
+void sub_08017B7C(u16 *buffer)
+{
+    struct SaveBlock1 *sb1 = gSaveBlock1Ptr;
+    u16 i = 0;
+
+    sb1->unk16[0] = buffer[0];
+    while (buffer[i])
+    {
+        ++i;
+        sb1->unk16[i] = buffer[i];
+    }
+}
+/* The structure of this func is the same to the upper one. But after checking the memory, unk16 is not where the name array lays. */
+
+void sub_08017BB4(u8 r0)
+{
+    gSaveBlock1Ptr->playerGender = r0;
+    sub_08012F0C(0x182,r0); // This apprears in main. Maybe I should figure out what this func does. 
```

Comment:
What do you mean by `Maybe I should figure out what this func does.`? We always want to figure out what every function does, right?

Reply:
Yes. I mean maybe I should figure out what this func does recently. ^_^